### DISCUSSION TITLE
How to build a full(er)-featured bootstrap jimsh?

### DISCUSSION THREAD

**Discussion by sgbeal** [msteveb/jimtcl#325] [Q&A]
This seemed straightforward enough but i'm failing time and again:

i'm attempting to create a full(er)-featured single-file jimsh.c by hacking on `make-bootstrap-jim`, starting small, like adding `namespace` support with a small change to:

```sh
cexts="namespace aio readdir regexp file exec clock array"
```

which ends in:

```console
$ ./test-bootstrap-jim 
Building bootstrap jimsh
jimsh_bootstrap.c:20336:12: error: redefinition of ‘Jim_EvalEnsemble’
20336 | static int Jim_EvalEnsemble(Jim_Interp *interp, const char *basecmd, const char *subcmd, int argc, Jim_Obj *const *argv)
      |            ^~~~~~~~~~~~~~~~
jimsh_bootstrap.c:2123:12: note: previous definition of ‘Jim_EvalEnsemble’ with type ‘int(Jim_Interp *, const char *, const char *, int,  Jim_Obj * const*)’
 2123 | static int Jim_EvalEnsemble(Jim_Interp *interp, const char *basecmd, const char *subcmd, int argc, Jim_Obj *const *argv)
      |            ^~~~~~~~~~~~~~~~
```

So instead...

```sh
allexts="bootstrap aio readdir regexp file glob exec clock array stdlib tclcompat namespace"
```

Which leads to:

```
$ ./test-bootstrap-jim 
Building bootstrap jimsh
/usr/bin/ld: /tmp/stephan/cceAdmeF.o: in function `Jim_InitStaticExtensions':
jimsh_bootstrap.c:(.text+0xa237): undefined reference to `Jim_namespaceInit'
collect2: error: ld returned 1 exit status
```

That function comes from `jim-namespace.c`, indicating that the namespace extension belongs in `cexts`, but doing that leads to ODR violations (yet not doing it leads to a binary without namespace support, as can be seen in the output from `test-bootstrap-jim`).

Certainly there's just some very simple trick i'm missing?

**Comment by msteveb** [comment]
There is a static Jim_EvalEnsemble() in both jim.c and jim-namespace.c. We can rename one of those.
